docs/docs/hydro/live-collections/streams.md

```diff
@@ -3,10 +3,119 @@ sidebar_position: 2
 ---
 
 # Streams
-You can view the full API documentation for Streams [here](pathname:///rustdoc/hydro_lang/stream/struct.Stream).
+Streams are the most common type of live collection in Hydro; they can be used to model streaming data collections, a feed of API requests, or even time-based intervals. A `Stream` represents a sequence of elements, with new elements being asynchronously appended to the end of the sequence. Streams can be transformed using APIs like `map` and `filter`, based on Rust [iterators](https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html). You can view the full API documentation for Streams [here](pathname:///rustdoc/hydro_lang/stream/struct.Stream).
+
+Streams have several type parameters:
+- `T`: the type of elements in the stream
+- `L`: the location the stream is on (see [Locations](../locations/index.md))
+- `B`: indicates whether the stream is [bounded or unbounded](./bounded-unbounded.md)
+- `Order`: indicates whether the elements in the stream have a deterministic order or not
+  - This type parameter is _optional_; by default the order is deterministic
+
+## Creating a Stream
+The simplest way to create a stream is to use the [`source_iter`](https://hydro.run/rustdoc/hydro_lang/location/trait.Location#method.source_iter) method on a location, which creates a stream from any Rust type that can be converted into an [`Iterator`](https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html) (via [`IntoIterator`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html)). For example, we can create a stream of integers on a [process](../locations/processes.md) and transform it:
+
+```rust
+# use hydro_lang::*;
+# use dfir_rs::futures::StreamExt;
+# tokio_test::block_on(test_util::multi_location_test(|flow, p_out| {
+let process = flow.process::<()>();
+let numbers: Stream<_, Process<_>, _> = process
+    .source_iter(q!(vec![1, 2, 3]))
+    .map(q!(|x| x + 1));
+// 2, 3, 4
+# numbers.send_bincode(&p_out)
+# }, |mut stream| async move {
+# for w in 2..=4 {
+#     assert_eq!(stream.next().await, Some(w));
+# }
+# }));
+```
+
+Streams also can be sent over the network to form distributed programs. Under the hood, sending a stream sets up an RPC handler at the target location that will receive the stream elements. For example, we can send a stream of integers from one process to another with [bincode](https://docs.rs/bincode/latest/bincode/) serialization:
```

Comment:
sent over the network to *participate in* distributed programs. (I don't think sending a stream forms a distributed program per se.)

Reply:
As regards the RPC handler, should we say something about the network protocol used underneath, e.g. whether it's TCP or UDP?